### ISSUE TITLE
Issues on python.org that are actively  being worked on or	closed remain in our list of 'open issues''

### ISSUE THREAD

**Issue by imported-from-roundup** [openhatch/oh-mainline#1221]
<b>Comment by <a href='https://openhatch.org/people/mandarg/'>mandarg</a>:</b><dl><dd>Example -- search for 'Volunteer Opportunities' by language 'Python'. 
[<a href="https://openhatch.org/search/?q=&amp;language=Python">https://openhatch.org/search/?q=&amp;language=Python</a>]

When I do this, the top five issues contain:
1. <a href="http://bugs.python.org/issue19669,">http://bugs.python.org/issue19669,</a> which is in Stage 'resolved'
2. <a href="http://bugs.python.org/issue19588,">http://bugs.python.org/issue19588,</a> which is in Stage 'patch review'
3. <a href="http://bugs.python.org/issue19626,">http://bugs.python.org/issue19626,</a> which is in Stage 'committed/rejected'

It seems to me that these decrease the signal-to-noise ratio a lot for new volunteers looking for 
open issues to work on. I would say we could restrict our issues to ones in Stage "needs patch" or 
"test needed".

An unrelated problem is also that the python bugtracker is possibly not always updated, for 
example, shouldn't <a href="http://bugs.python.org/issue1100942">http://bugs.python.org/issue1100942</a> be in Stage "patch-review"? We could 
also get around this by excluding all issues with a patch attached.

</dl></dd><hr/>

<b>Comment by <a href='https://openhatch.org/people/ehashman/'>ehashman</a>:</b><dl><dd>I'll take a look at this. Sounds like it should be a quick and easy fix.

</dl></dd><hr/>

<b>Comment by <a href='https://openhatch.org/people/ehashman/'>ehashman</a>:</b><dl><dd>Verified that the bug still exists (this bug comes up in a search even though
it's in review: <a href="http://bugs.python.org/issue9305">http://bugs.python.org/issue9305</a> and this one is
committed/rejected: <a href="http://bugs.python.org/issue20145">http://bugs.python.org/issue20145</a>).

Fix for this is to change "closed" to "closed,resolved,patch
review,committed/rejected" in the Python tracker configuration in
openhatch.org/customs, which I've done. I've also added "commit review" to that
based on this bug coming up in search: <a href="http://bugs.python.org/issue20627">http://bugs.python.org/issue20627</a> 

This config change will cause the affected bugs to be filtered out from search
(see bugimporters/roundup.py:201 in oh-bugimporters and
mysite/search/models.py:560 in oh-mainline).

Someone should verify that this is working from the next bugimporters crawl.

Possible followup: it appears that python has marked these bugs
(<a href="http://bugs.python.org/issue20145,">http://bugs.python.org/issue20145,</a> <a href="http://bugs.python.org/issue20627">http://bugs.python.org/issue20627</a>) as
closed, but not in their status. Filtering out the requested statuses as per
this fix will probably take care of the majority of closed bugs, but perhaps
this other closed field should be investigated.

</dl></dd><hr/>

<b>Comment by <a href='https://openhatch.org/people/ehashman/'>ehashman</a>:</b><dl><dd>I've refiled the second half of the bug report as <a href="issue953">issue 953</a>:
<a href="https://openhatch.org/bugs/issue953">https://openhatch.org/bugs/issue953</a>

</dl></dd><hr/>

<b>Comment by <a href='https://openhatch.org/people/ehashman/'>ehashman</a>:</b><dl><dd>I accidentally put the status change on the wrong comment, sorry! :)

On 2014-04-04 09:14, ehashman wrote:
&gt; ehashman &lt;<a href="mailto:elana@hashman.ca">elana@hashman.ca</a>&gt; added the comment:
&gt;
&gt; I've refiled the second half of the bug report as <a href="issue953">issue 953</a>:
&gt; <a href="https://openhatch.org/bugs/issue953">https://openhatch.org/bugs/issue953</a>
&gt;
&gt; ----------
&gt; status: chatting -&gt; need-review
&gt;
&gt; __________________________________________
&gt; Roundup issue tracker &lt;<a href="mailto:bugs@openhatch.org">bugs@openhatch.org</a>&gt;
&gt; &lt;<a href="https://openhatch.org/bugs/issue906">https://openhatch.org/bugs/issue906</a>&gt;
&gt; __________________________________________
&gt; _______________________________________________
&gt; Devel mailing list
&gt; <a href="mailto:Devel@lists.openhatch.org">Devel@lists.openhatch.org</a>
&gt; <a href="http://lists.openhatch.org/mailman/listinfo/devel">http://lists.openhatch.org/mailman/listinfo/devel</a>

</dl></dd><hr/>

<b>Comment by <a href='https://openhatch.org/people/ehashman/'>ehashman</a>:</b><dl><dd>So it's been a couple days and I haven't seen any changes take place. There seem
to be two issues here:

1) Bugs that are marked "Status: closed" are not being removed from our bug
collection. There's lots of them that have been closed for a while (1+ months)
but we still display. This may be a separate issue.

2) I think I've confused the two separate fields of "Status" vs. "Stage". We are
pulling the former, not the latter (oops -.-). It shouldn't be too painful to
grab the stage info; indeed, it should already be present in the metadata
dictionary from the parse. (bugimporters/roundup.py:111) So the question then is
how to deal with this specifically in the Python tracker case without making any
groundbreaking changes---it would be nice to not have to add a configuration
setting for a stage, e.g.

I propose to replace the "Status" value with the "Stage" value if "Status" ==
open. This has the added benefit of better UX for those browsing in our
volunteer opportunities search, as they get more information about the bug
status at a glance without having to open the issue. To accomplish this, all
that needs to be done is an override of a small piece of functionality in
get_parsed_data_dict for the roundup trackers (line 197).

</dl></dd><hr/>

<b>Comment by <a href='https://openhatch.org/people/ehashman/'>ehashman</a>:</b><dl><dd>Fix complete. Opened pull request:
<a href="https://github.com/openhatch/oh-bugimporters/pull/46">https://github.com/openhatch/oh-bugimporters/pull/46</a>

</dl></dd><hr/>

<b>Comment by <a href='https://openhatch.org/people/ehashman/'>ehashman</a>:</b><dl><dd>Pull request has been merged; 
however, apparently Travis didn't 
like it very much. Going to leave 
this open another few days to verify 
it hasn't broken the bugimporters 
before closing this issue.

</dl></dd><hr/>

<b>Comment by <a href='https://openhatch.org/people/ehashman/'>ehashman</a>:</b><dl><dd>Changed custom parser to PythonRoundupBugParser for Python. This shouldn't break
anything. Waiting on <a href="https://github.com/openhatch/oh-bugimporters/pull/49">https://github.com/openhatch/oh-bugimporters/pull/49</a> aka
<a href="issue962">issue962</a> for the change to take effect.

</dl></dd><hr/>

Status: <b>resolved</b>
Nosy List: <b>  ehashman, mandarg, paulproteus </b>
Priority: <b>urgent</b>
Imported from roundup ID: <b>906</b> (<a href='http://roundup-archive.openhatch.org/bugs/issue906'>view archived page</a>)
Last modified: <b>2014-05-04.22:31:38</b>
